cmd/prepare-storage.go

```diff
@@ -169,17 +168,6 @@ func prepForInitXL(firstDisk bool, sErrs []error, diskCount int) InitActions {
 		return WaitForFormatting
 	}
 
-	// Total disks unformatted are in quorum verify if we have some offline disks.
-	if disksUnformatted >= quorum {
-		// Some disks offline and some disks unformatted, wait for all of them to come online.
-		if disksUnformatted+disksFormatted+disksOffline == diskCount {
-			return WaitForAll
-		}
-
-		// Some disks possibly corrupted and too many unformatted disks.
-		return Abort
-	}
-
 	// Already formatted and in quorum, proceed to initialization of object layer.
 	if disksFormatted >= readQuorum {
 		if disksFormatted+disksOffline == diskCount {
```

Comment:
In this block of code, we proceed to initialize object layer only if disks are either formatted or offline. But we should initialize object layer and serve GET/HEAD requests on buckets and objects.

Observe what happens in a 8 disk setup where,
```go
  disksFormatted = 4
  disksUnformatted = 2
  disksOffline = 2
  readQuorum = 4
```
With this change, we will wait until one or more disks to be healed or come on-line.

To avoid this we should,
```go
if disksFormatted >= readQuroum {
  if disksFormatted + disksUnformatted + disksOffline == diskCount {
    return InitObjectLayer  
  }
}
```

Reply:
This case is already handled.